intelligence/ts/src/flowerintelligence.ts

```diff
@@ -207,19 +207,37 @@ export class FlowerIntelligence {
       // Currently we just select the first compatible localEngine without further check
       return { ok: true, value: compatibleEngines[0] };
     } else {
+      let description = `No available local engine for \`${modelId}\``;
+      description += isNode
+        ? ' (make sure you have enough available RAM).'
+        : ' (make sure you are using a WebGPU compatible browser and have enough available RAM).';
       return {
         ok: false,
         failure: {
           code: FailureCode.NoLocalProviderError,
-          description: `No available local engine for ${modelId}.`,
+          description,
         },
       };
     }
   }
 
   private getOrCreateRemoteEngine(localFailure?: Result<Engine>): Result<Engine> {
-    if (localFailure && !FlowerIntelligence.#remoteHandoff && !FlowerIntelligence.#apiKey) {
-      return localFailure;
+    if (
+      localFailure &&
+      !localFailure.ok &&
+      (!FlowerIntelligence.#remoteHandoff || !FlowerIntelligence.#apiKey)
+    ) {
+      let description = localFailure.failure.description;
+      description += FlowerIntelligence.#remoteHandoff
+        ? '\nAnd no valid API key was provided for Remote Handoff.'
+        : '\nAnd Remote Handoff was not enabled with a valid API key.';
```

Comment:
```suggestion
        : '\nAdditionally, Remote Handoff with a valid API key was not enabled.';
```

Reply:
@tanertopal Maybe we can remove the API key part in this case:
```suggestion
        : '\nAdditionally, Remote Handoff was not enabled.';
```

Reply:
@charlesbvll I just had a sync with Daniel. Let's stick to a single description per error code and link to the documentation for more details. Basically the same solution we have for the Flower Framework. Can make the adjustments?